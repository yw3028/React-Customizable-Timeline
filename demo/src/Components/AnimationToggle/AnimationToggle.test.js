import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import userEvent from '@testing-library/user-event';
import AnimationToggle from './AnimationToggle';

const renderComponent = animation => render(<AnimationToggle isAnimated={animation} />);

describe('AnimationToggle component', () => {
  afterEach(() => {
    cleanup();
  });

  test('render correctly', async () => {
    const component = renderComponent(true);
    expect(component.baseElement).toMatchSnapshot();
  });

  it('Renders the state', async () => {
    renderComponent(true);
    expect(screen.queryByText('ON')).toBeTruthy();
    expect(screen.queryByText('OFF')).toBeFalsy();
    cleanup();
    renderComponent(false);
    expect(screen.queryByText('ON')).toBeFalsy();
    expect(screen.queryByText('OFF')).toBeTruthy();
  });

  it("Doesn't render children elements", async () => {
    render(
      <AnimationToggle isAnimated={false}>
        <p>LisaPisa</p>
      </AnimationToggle>,
    );

    await expect(screen.queryByText(/LisaPisa/i)).toBeFalsy();
  });

  it('Changes the state', () => {
    renderComponent(true);
    userEvent.click(screen.getByText('ON'));
    expect(screen.getByText('OFF')).toBeTruthy();
  });
});
