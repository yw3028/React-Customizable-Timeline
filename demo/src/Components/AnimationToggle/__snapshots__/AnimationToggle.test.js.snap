// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimationToggle component render correctly 1`] = `
<body>
  <div>
    <button
      class="toggle-button"
    >
      ON
    </button>
  </div>
</body>
`;
